src/Core/MapUtil/FHTileMap.cpp

```diff
@@ -690,6 +690,89 @@ void FHTileMap::correctTerrainTypes(Core::LibraryTerrainConstPtr dirtTerrain,
     });
 }
 
+void FHTileMap::clearRoads(std::vector<FHTown> m_towns)
+{
+    for (bool reverseOrder : {false, true}) {
```

Comment:
that is only do 2 transforms (base and diagonal mirror) 
i'd better do simplification of conditions but do full pattern scan (as it currently done for terrain patterns). 
not like 4 mirrors but rather do transpose + mirror logic (so you don't need to do pattern matching for horizontal deadend and vertical deadend separetely)